lib/src/device_scanner.dart

```diff
@@ -48,13 +50,16 @@ class DeviceScanner {
     final completer = Completer<void>();
     _currentScanSession =
         ScanSession(withServices: withServices, future: completer.future);
+    // we do not want the future to throw the error but we are going to throw it in the Repeater
+    unawaited(completer.future.catchError((Object e, StackTrace s) {}));
 
     final scanRepeater = Repeater(
       onListenEmitFrom: () =>
           _scanStream.map((scan) => scan.result.dematerialize()).handleError(
-        (Object e, StackTrace s) {
+        (Exception e, StackTrace s) {
```

Comment:
How can you require an `Exception` here, if the contract is "anything" (correct me, if I'm wrong)? This way you can only get a false sense of type-safety, because you can still get an `Error` or anything at all here.

I'd keep the parameter type `Object` and rethrow it somehow like: `if (e is Exception) throw e; else if (e is Error) throw e; else throw Exception(e);`

Note that a shorter `e is Exception ? e : e is Error ? e : Exception(e)` wouldn't work, because `Exception` and `Error` have no subclass-baseclass relationship (they both inherit from `Object` only), thus the type of the entire expression would be `Object` and a warning would pop-up.

Alternatively, a directive can be used to disable the corresponding diagnostic, but... I'd first looks into solutions within the bounds of the language.

Reply:
ah indeed I forgot about that good catch!